Pod/Classes/VOKCoreDataCollectionTypes.h

```diff
@@ -0,0 +1,61 @@
+//
+//  VOKCoreDataCollectionTypes.h
+//  Pods
+//
+//  Created by Carl Hill-Popper on 10/23/15.
+//
+//
+
+#ifndef VOKCoreDataCollectionTypes_h
+#define VOKCoreDataCollectionTypes_h
+
+#if __has_feature(objc_generics)
+//forward declare classes used here since we don't know where this will be included
+@class NSManagedObject;
+@class NSManagedObjectID;
+@class VOKManagedObjectMap;
+
+//include the pointer for this one since the fallback is id
+#define VOKManagedObjectSubclassPtr __kindof NSManagedObject *
```

Comment:
Array and dictionary types like this don't accept subclasses by default?  Is there any other place in ObjC where that's the case?  Is the same true in Swift?